release/ray_release/tests/test_cluster_manager.py

```diff
@@ -100,26 +92,28 @@ def setUp(self) -> None:
             test=Test(
                 {
                     "name": f"unit_test__{self.__class__.__name__}",
-                    "cluster": {},
+                    "cluster": {"byod": {}},
                 }
             ),
         )
         self.sdk.reset()
         self.sdk.returns["get_cloud"] = APIDict(result=APIDict(provider="AWS"))
+        os.environ["BUILDKITE_COMMIT"] = "123456"
+        os.environ["BUILDKITE_BRANCH"] = "master"
```

Comment:
why are these needed?

Reply:
they are used for byod tests to name the byod docker image: https://github.com/ray-project/ray/blob/master/release/ray_release/test.py#L224; previously those tests didn't need to set these because it never execute that parts of the code ;)

Reply:
I worry that this changes the env permanently without changing back, and will make other tests harder.

could you:

- set up the env early, like even before `self.sdk = MockSDK()`
- saves the original env state of these keys.
- add a `tearDown()` function, and restore the env to its original state.

alternatively, you can also support some kind of env lookup function pointer, and send it into Test on construction, and load the env with that function. similar to what I did in rayci in go:

https://github.com/ray-project/rayci/blob/main/raycicmd/env.go#L15